### PR TITLE
RavenDB-4474 Add a feature to set and get thread priority

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: src/Sparrow/Platform/ThreadMethods.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                {
+                    int lastWin32ErrorCode = Marshal.GetLastWin32Error();
+                    throw new Win32Exception("Failed to set priority to thread " + threadId,
+                        new Win32Exception(lastWin32ErrorCode));
+                }
+            }
+            else
+            {
+                uint threadId = Win32ThreadsMethods.GetCurrentThreadId();
+                IntPtr handle = Win32ThreadsMethods.OpenThread(ThreadAccess.SetInformation, false, threadId);
```
Comment (ayende) [inline src/Sparrow/Platform/ThreadMethods.cs]: Need error handling here. This call may fail

File: test/FastTests/Sparrow/ChangeThreadPriorityTest.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+﻿using System.Threading;
+using Sparrow;
+using Xunit;
+
+namespace FastTests.Sparrow
+{
+    public class ChangeThreadPriorityTest
+    {
+        [Fact]
+        public static void StartChangeThreadPriority()
```
Comment (ayende) [inline test/FastTests/Sparrow/ChangeThreadPriorityTest.cs]: Test methods shouldn't be static

File: test/FastTests/Sparrow/ChangeThreadPriorityTest.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        {
+            Thread first = new Thread(() => ChangeThreadPriorityCheck("First"));
+            Thread second = new Thread(() => ChangeThreadPriorityCheck("Second"));
+            Thread third = new Thread(() => ChangeThreadPriorityCheck("Third"));
+
+            first.Start();
```
Comment (ayende) [inline test/FastTests/Sparrow/ChangeThreadPriorityTest.cs]: you aren't waiting for the threads to complete

File: test/FastTests/Sparrow/ChangeThreadPriorityTest.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            second.Start();
```
Comment (ayende) [inline test/FastTests/Sparrow/ChangeThreadPriorityTest.cs]: An error here will take down the entire test suite, shouldn't be the case.

File: src/Sparrow/Platform/ThreadMethods.cs (hunk 1 above)
Comment (IdanHaim) [inline src/Sparrow/Platform/ThreadMethods.cs]: The pthread_setschedparam always return zero or none zero in the case the method fail

Comment (ayende) [inline src/Sparrow/Platform/ThreadMethods.cs]: I'm actually looking at the `OpenThread` error handling

Comment (IdanHaim) [inline src/Sparrow/Platform/ThreadMethods.cs]: Yea I figure it up

**Comment by ppekrol** [comment]
[15:25:11][Step 1/1] Detected tabs in 'src/Sparrow/Platform/Posix/PosixThreadsMethods.cs'
[15:25:11][Step 1/1] Detected tabs in 'src/Sparrow/Platform/ThreadMethods.cs'

**Pull request merged**